webrender/res/prim_shared.glsl

```diff
@@ -222,14 +223,21 @@ VertexInfo write_transform_vertex(RectWithSize local_segment_rect,
     return vi;
 }
 
-void write_clip(vec4 world_pos, ClipArea area) {
-    vec2 uv = world_pos.xy * uDevicePixelRatio +
+void write_clip(vec4 world_pos, vec2 snap_offset, ClipArea area) {
+    vec2 uv = snap_offset + world_pos.xy * uDevicePixelRatio +
```

Comment:
snap_offset needs to be multiplied by `world_pos.w` to be homogeneous

Reply:
Fixed